### ISSUE TITLE
Percent (`%`) as first line when I start Alacritty

### ISSUE THREAD

**Issue by predmijat** [jwilm/alacritty#1616]
Which operating system does the issue occur on?
Arch Linux

If on linux, are you using X11 or Wayland?
X11

Hi,

When I run Alacritty on Linux, I get percent sign (`%`) as the first line, then the blank line, and then my prompt. It happens with default zshrc too. Terminator starts fine (first line is my prompt). Alacritty on macOS with same zshrc starts fine.

![2018-10-02_21-11](https://user-images.githubusercontent.com/8684411/46371319-52b14000-c688-11e8-82d3-40627e51265f.png)

Thanks

**Comment by chrisduerr** [comment]
Can you consistently reproduce this? I've run into it a couple of times but it only seems to appear occasionally when the shell takes a lot of time to start.

**Comment by predmijat** [comment]
Yes, every single time when I start it...

**Comment by chrisduerr** [comment]
Hmm that's interesting. Is this related to the shell in use in any way? Or does this happen with every shell?

Also note that this doesn't seem to be a normal `%`, but rather the same symbol you get when you end a line without a newline.

**Comment by predmijat** [comment]
Yeah I think it's a newline thing too. I just changed my default shell to bash, ran Alacritty and I can see `%` for a second, but it quickly disappears and prompt is shown as first line.

**Comment by utkuboduroglu** [comment]
I run zsh with Alacritty on i3 and this also happens to me. I found that it happens because of two things:
1. I run tmux while launching Alacritty, and if I have more than 1 tmux window,
2. Because of i3, whenever my Alacritty window is resized while launching.

**Comment by kchibisov** [comment]
Seems like this bug is related to #2817 , since this bug is also caused by multiple resizes during startup.

**Comment by minhduc0711** [comment]
This occured to me as well, but only on my fractionally scaled monitor. I guess resizing is the correct cause.

**Comment by lisuke** [comment]
```
OS:        Arch Linux
WM:        Hyprland
SHELL:     Zsh
```
![图片](https://github.com/alacritty/alacritty/assets/10797357/c9182620-e4a0-4099-aa85-444a382c20f0)

**Comment by kchibisov** [comment]
@lisuke it's a new line character from zsh. You could try the latest master of alacritty, it should be a bit better when it comes to resizes on Wayland, so the issue might be avoided. Or just delay the startup of the shell.

**Comment by frogpal** [comment]
I've had the same issue, but after some [googling](https://superuser.com/a/1391570), it turns out that it might be a problem related to how zsh treats outputs without proper line endings. So if you have a script in your `.zshrc` that outputs characters without a trailing newline during startup, you might see the percentage sign on the first line. 

One workaround that I found is to add `unsetopt PROMPT_SP` in your `.zshrc` to prevent zsh from adding percentage signs on outputs without trailing newline characters. You lose the incomplete line detection for actual commands, but many people disable this anyway.

So you can just do:

```sh
echo "unsetopt PROMPT_SP" >> ~/.zshrc
```

and the "%" should be gone from the first line on startup. 

I think you might consider closing the issue now @chrisduerr
